query/src/main/java/org/infinispan/query/impl/massindex/IndexWorker.java

```diff
@@ -67,42 +61,38 @@
+      boolean valueFilterable = valueWrapper.isFilterable();
 
-      SearchIntegrator searchIntegrator = ComponentRegistryUtils.getSearchIntegrator(unwrapped);
-      KeyTransformationHandler keyTransformationHandler = ComponentRegistryUtils.getKeyTransformationHandler(unwrapped);
-      TimeService timeService = ComponentRegistryUtils.getTimeService(unwrapped);
+      AdvancedCache<Object, Object> reindexCache = valueFilterable ? cache.withStorageMediaType() : cache;
+
+      SearchIntegrator searchIntegrator = ComponentRegistryUtils.getSearchIntegrator(cache);
+      KeyTransformationHandler keyTransformationHandler = ComponentRegistryUtils.getKeyTransformationHandler(cache);
+      TimeService timeService = ComponentRegistryUtils.getTimeService(cache);
 
       IndexUpdater indexUpdater = new IndexUpdater(searchIntegrator, keyTransformationHandler, timeService);
-      ClusteringDependentLogic clusteringDependentLogic = SecurityActions.getClusteringDependentLogic(unwrapped);
+      ClusteringDependentLogic clusteringDependentLogic = SecurityActions.getClusteringDependentLogic(cache);
       KeyPartitioner keyPartitioner = ComponentRegistryUtils.getKeyPartitioner(cache);
 
-      DataConversion keyDataConversion = unwrapped.getKeyDataConversion();
-      DataConversion valueDataConversion = unwrapped.getValueDataConversion();
-
       if (keys == null || keys.size() == 0) {
          preIndex(indexUpdater);
          if (!skipIndex) {
-            KeyValueFilter filter = getFilter(clusteringDependentLogic, keyDataConversion);
-            try (Stream<CacheEntry<Object, Object>> stream = cache.getAdvancedCache().withFlags(Flag.CACHE_MODE_LOCAL)
+            KeyValueFilter filter = getFilter(clusteringDependentLogic, reindexCache.getKeyDataConversion());
+            try (Stream<CacheEntry<Object, Object>> stream = reindexCache.getAdvancedCache().withFlags(Flag.CACHE_MODE_LOCAL)
                   .cacheEntrySet().stream()) {
                Iterator<CacheEntry<Object, Object>> iterator = stream.filter(CacheFilters.predicate(filter)).iterator();
-               Wrapper wrapper = unwrapped.getValueDataConversion().getWrapper();
                while (iterator.hasNext()) {
                   CacheEntry<Object, Object> next = iterator.next();
-                  Object value = extractValue(next.getValue(), valueDataConversion);
-                  Object storedKey = keyDataConversion.toStorage(next.getKey());
-                  int segment = keyPartitioner.getSegment(storedKey);
-                  if (value instanceof byte[] && storageType != OBJECT) {
-                     value = wrapper.wrap(value);
+                  Object key = next.getKey();
+                  Object storedKey = reindexCache.getKeyDataConversion().toStorage(key);
+                  Object value = next.getValue();
+                  // TODO: Way to grab a cache without unwrapping? cache.withStoreMediaType(boolean unwrap)
```

Comment:
Is this for this PR?

Reply:
let me check if this is still needed

Reply:
For now, I will just remove this TODO and handle it in the next PR